sdk/identity/src/token_credentials/client_secret_credentials.rs

```diff
@@ -4,9 +4,55 @@ use chrono::Utc;
 use oauth2::{
     basic::BasicClient, reqwest::async_http_client, AccessToken, AuthType, AuthUrl, Scope, TokenUrl,
 };
-use std::{str, time::Duration};
+use std::{borrow::Cow, str, time::Duration};
 use url::Url;
 
+/// Provides options to configure how the Identity library makes authentication
+/// requests to Azure Active Directory.
+#[derive(Clone, Debug, PartialEq)]
+pub struct TokenCredentialOptions {
+    /// The authority host to use for authentication requests.  The default is
+    /// "https://login.microsoftonline.com".
+    pub authority_host: Option<Cow<'static, str>>,
```

Comment:
I think `Cow` here is unnecessary, the `'static` lifetime makes owning the string redundant.

Reply:
My goal was to be able to set an owned `String` or a `'static str'. I can provide examples tomorrow. May be I should add tests that demo this.

Reply:
Ah yes, if we want to support custom authorities `Cow` makes sense. 
But then I would relax the `'static` requirement, it might be useful to pass a short lived &str.

Reply:
If this stays public (which I think it shouldn't) then we should just make this an owned `String` and require users to turn string slices into owned `String`s. Otherwise, I think the `Cow` is fine as an implementation detail. If users want to provide a completely custom authority they can always provide it as a `String`.

Reply:
I just made it a `String` to keep it simple.